src/Agent/Communication/AbstractRequest.php

```diff
@@ -305,15 +305,23 @@ public function getMode()
     /**
      * Adds an error
      *
-     * @param string $message Error message
+     * @param string  $message Error message
+     * @param integer $code    HTTP response code
      *
      * @return void
      */
     public function addError($message, $code = 500)
     {
         $this->error = true;
-        $this->http_repsonse_code = $code;
-        if ($this->headers->hasHeader('GLPI-Agent-ID')) {
+        $this->http_response_code = $code;
+        if ($code == 400 && preg_match('/not supported/', $message)) {
+            $this->setMode(self::JSON_MODE);
+            $this->addToResponse([
+            'status' => 'error',
+            'message' => $message,
+            'expiration' => self::DEFAULT_FREQUENCY
+            ]);
```

Comment:
Seems not really generic. Maybe this piece of code should be moved to the specific handling of `getConfig`.

Reply:
addError() is made more generic with the latest 2 commits.